### PR TITLE
Update colors based on latest swatches

### PR DESCRIPTION
**Pull request by taysea**
### In this PR
This updates color values/names as well as tab styling (since some is reliant on these values). I am updating everything that has a 1:1 mapping or could easily added but outstanding issues are noted at the bottom of this PR.

The values are coming from here: https://www.figma.com/file/eZYR3dtWdb9U90QvJ7p3T9/HPE-Color-Styles?node-id=590%3A981

### Items that need to be addressed separately:
- [ ] How will these new validation colors play out in a form? I believe it will require some grommet changes, but I'm curious if the styles will be opt-in depending on the form or if they'll automatically be applied: 
Screen Shot 2020-04-21 at 3 56 20 PM

- [ ] formField.disabled.background has not been updated. Currently, global.opacity.medium is being applied to disabled formFields, so applying a color of `#0000000f` gets over-ridden by rgba(0,0,0,0.4) in the DOM. I haven't filled in this value because it will make disabled fields too dark. Need to look into fix more.
- [ ] Control colors of primary/secondary need to be discussed a little more -- are these just for Button or will this apply to other control components like menu, etc. where a control color is sometimes used?

### CODE REVIEW

File: src/js/index.js [hunk 1]
```diff
@@ -70,45 +78,51 @@ export const hpe = deepFreeze({
...
-        light: '#FFB024',
+        dark: '#9B6310',
+        light: '#FFBC44',
       },
       'orange!': '#FF8300',
       yellow: {
-        dark: '#D78F00',
-        light: '#FFEB59',
+        dark: '#8D741C',
+        light: '#FFD63E',
```
Comment (halocline) [inline src/js/index.js:125]: Looks like this has shifted: https://www.figma.com/file/eZYR3dtWdb9U90QvJ7p3T9/HPE-Color-Styles?node-id=590%3A1231
```suggestion
        light: '#FFEB59',
```

File: src/js/index.js [hunk 2]
```diff
@@ -286,7 +300,7 @@ export const hpe = deepFreeze({
     },
     error: {
       background: {
-        color: { light: '#FF404033', dark: '#FF40404D' },
+        color: { light: '#FC6161', dark: '#C54E4B' },
```
Comment (halocline) [inline src/js/index.js:303]: These look suspect. They are the colors of 'red' and 'red!', which are correct-ish. The visual color has these colors plus a 24% and 36% opacity. Ideally we would reference by name rather than use HEX here. Potentially could have a function which receives (colorname, opacity) and returns a hexvalue with opacity.

```suggestion
        color: { light: '#FC61613D', dark: '#C54E4B5C' },
```

File: src/js/index.js [hunk 3]
```diff
@@ -513,16 +527,31 @@ export const hpe = deepFreeze({
...
-      horizontal: 'small',
+    border: {
+      side: 'bottom',
+      color: 'background-back',
+      active: {
+        color: 'border',
+      },
+      hover: {
+        color: 'background-back',
+      },
```
Comment (halocline) [inline src/js/index.js:546]: Seems like these may have changed. In https://www.figma.com/file/Kp4dWyhUTnKIJ1Cg5CoL9o/HPE-Tabs-Component?node-id=208%3A0 I am seeing the following:

```suggestion
      color: 'border',
      active: {
        color: 'border-strong',
      },
      hover: {
        color: 'border',
      },
```

File: src/js/index.js (hunk 2 above)
Comment (halocline) [reply src/js/index.js:303]: This may be related to your commentary in the PR description, but wasn't exactly sure.

File: src/js/index.js [hunk 4]
```diff
@@ -513,16 +527,31 @@ export const hpe = deepFreeze({
...
+      hover: {
+        color: 'border',
+      },
     },
+    pad: 'small',
+    margin: 'none',
+    extend: ({ theme }) => css`
+      border-top-left-radius: ${theme.global.control.border.radius};
+      border-top-right-radius: ${theme.global.control.border.radius};
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/index.js:552]: Sounds good -- is that something we need to implement in grommet prior to merging this PR? or something that can be added in after and then updated?

[the remark this answers is not in the document]
Comment (ShimiSun) [reply src/js/index.js:552]: we can add it later. for now:
```suggestion
      border-top-left-radius: ${theme.global.control.border.radius}; // should use radius property of border
      border-top-right-radius: ${theme.global.control.border.radius}; // should use radius property of border
```

File: src/js/index.js [hunk 5]
```diff
@@ -27,38 +27,46 @@ export const hpe = deepFreeze({
...
+        light: '#999999',
+      },
+      'border-strong': {
+        dark: '#AFBCD2',
         light: '#666666',
       },
+      'border-weak': {
+        dark: '#606B7D',
+        light: '#BBBBBB',
+      },
```
Comment (ShimiSun) [inline src/js/index.js:69]: I commented it earlier and didn't see a direct reply so reposting
>Just to confirm, were the new colors tested on the application side prior to the PR submission? Just to be sure we are not introducing new contrast issues.

Comment (taysea) [reply src/js/index.js:69]: Bringing over a discussion from Slack:
Using text color of `text` on the dark colors doesn’t meet contrast requirements, but changing the text color to `text-strong` passes everything. Feels like the site will need some guidance of when to use text vs text-strong to avoid the first error

`text` on dark color palette:
Screen Shot 2020-04-23 at 10 04 59 AM

`text-strong` on dark color palette:
Screen Shot 2020-04-23 at 10 13 01 AM

Comment (taysea) [reply src/js/index.js:69]: As long as we provide appropriate guidance of how to use text colors, then we are fine from a color contrast perspective.

File: src/js/index.js [hunk 6]
```diff
@@ -513,16 +527,31 @@ export const hpe = deepFreeze({
...
+    hover: {
+      background: 'background-contrast',
+      color: 'text',
     },
-    pad: {
-      horizontal: 'small',
+    border: {
+      side: 'bottom',
+      color: 'border',
+      active: {
```
Comment (ericsoderberghp) [inline src/js/index.js:541]: On the grommet side, this should be `active.border` instead of `border.active` to align with the theme structure direction. Not an issue for this PR per-se.

File: src/js/index.js [hunk 7]
```diff
@@ -513,16 +527,31 @@ export const hpe = deepFreeze({
...
+        color: 'border-strong',
+      },
+      hover: {
+        color: 'border',
+      },
     },
+    pad: 'small',
+    margin: 'none',
+    extend: ({ theme }) => css`
```
Comment (ericsoderberghp) [inline src/js/index.js:550]: Would be nice to not need `extend` for HPE theme. Not part of this PR per-se, but something we should look into. Can we add an issue to evaluate the use of `extend` in the HPE NEXT theme, to see if anything could/should be easier.
